### DISCUSSION TITLE
Extract character from matrix variable

### DISCUSSION THREAD

**Discussion by reikdas** [community/community#26350] [Actions]
<p>I have a matrix variable setup like this -</p>
<pre><code class="lang-auto">strategy:
      matrix:
        compiler: [gcc-5, gcc-6, gcc-8, gcc-9, clang-4, clang-7, clang-8, clang-9, clang-10, clang-11]
</code></pre>
<p>How do I extract the numbers from this?</p>
<p>I tried doing -</p>
<pre><code class="lang-auto">if [ "${{ matrix.compiler }}" = *"gcc"* ]; then
    echo "CC=gcc-${{ matrix.compiler }:4:1}" &gt;&gt; $GITHUB_ENV
    echo "CXX=g++-${{ matrix.compiler }:4:1}" &gt;&gt; $GITHUB_ENV
else
    echo "CC=clang-${{ matrix.compiler }:5:1}" &gt;&gt; $GITHUB_ENV
    echo "CXX=clang++${{ matrix.compiler }:5:1}" &gt;&gt; $GITHUB_ENV
fi
</code></pre>
<p>where I try to get the 4th and the 5th character from the string as <code>${{ matrix.compiler }:4:1}</code> and <code>${{ matrix.compiler }:5:1}</code> but that doesn’t work :frowning:</p>

**Comment by bsa-coder** [comment]
<p>I’m new at GitHub Actions so I don’t know the exact answer; but, I did a little research and …</p>
<p>You can’t use BASH commands/logic within your YML workflow.</p>
<p>StackOverflow (<a href="https://stackoverflow.com/questions/60916931/github-action-does-the-if-have-an-else" rel="noopener nofollow ugc">https://stackoverflow.com/questions/60916931/github-action-does-the-if-have-an-else</a>) says there is no IF-THEN-ELSE.</p>
<p>If is actions is of the form (got this from (<a href="https://github.community/t/run-step-on-if-branch-tag-is/16965/2" class="inline-onebox">run step on (if branch/tag is)</a>)</p>
<p>if: contains(github.ref, “release”)<br>
steps:</p>
<ul>
<li>run: echo “I only run if the branch has release in its name!”</li>
</ul>
<p>You’ll need to restructure your logic … I think.</p>
<p>This link has lots of useful documentation(<a href="https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions" rel="noopener nofollow ugc">https://docs.github.com/en/free-pro-team@latest/actions/reference/workflow-syntax-for-github-actions</a>)</p>

**Comment by shivammathur** [comment, marked as answer]
<p>Store the compiler value from matrix as an env variable and use that.</p>
<pre><code class="lang-auto">- name: Set compiler    
  run: |
    if [[ "$compiler" =~ ^gcc* ]]; then
      echo "CC=gcc-${compiler:4}" &gt;&gt; $GITHUB_ENV
      echo "CXX=g++-${compiler:4}" &gt;&gt; $GITHUB_ENV
    else
      echo "CC=clang-${compiler:6}" &gt;&gt; $GITHUB_ENV
      echo "CXX=clang++-${compiler:6}" &gt;&gt; $GITHUB_ENV
    fi
  env:
    compiler: ${{ matrix.compiler }}
</code></pre>
